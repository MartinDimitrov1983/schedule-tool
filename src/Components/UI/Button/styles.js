export function styles(color) {
  return {
    background: color,
    borderRadius: 0,
    width: '100%',
    textTransform: 'capitalize',
    boxShadow: 'none',
    fontWeight: 'bold',
    '&:hover': {
      background: color,
      boxShadow: 'none',
    },
    '&.Mui-disabled': {
      color: 'white',
      background: color,
      opacity: 0.35,
    },
  };
}
