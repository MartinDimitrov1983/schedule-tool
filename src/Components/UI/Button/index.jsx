import React from 'react';
import { Button as MuiButton } from '@mui/material';
import PropTypes from 'prop-types';
import { styles } from './styles';

const Button = ({
  text,
  disabled,
  color,
  onClick,
  onMouseEnter,
  onMouseLeave,
}) => {
  return (
    <MuiButton
      variant="contained"
      disabled={disabled}
      sx={styles(color)}
      onClick={onClick}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {text}
    </MuiButton>
  );
};

Button.propTypes = {
  text: PropTypes.string,
  disabled: PropTypes.bool,
  color: PropTypes.string,
  onClick: PropTypes.func,
};

export default Button;
