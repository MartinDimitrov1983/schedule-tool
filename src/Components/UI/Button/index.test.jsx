import React from 'react';
import { render } from '@testing-library/react';

import Button from './index';

test('Snapshot Button', () => {
  const { asFragment } = render(
    <Button
      text="Test"
      disabled={false}
      color="#E11BB6"
      onClick={() => {}}
      onMouseEnter={() => {}}
      onMouseLeave={() => {}}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
