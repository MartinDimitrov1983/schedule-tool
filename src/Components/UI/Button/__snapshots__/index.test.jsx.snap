// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Button 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1m39mvv-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Test
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
